src/components/document/document-content.tsx

```diff
@@ -67,16 +70,37 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
     this.mutationObserver.disconnect();
   }
 
-  public componentDidUpdate() {
+  public componentDidUpdate(prevProps: IProps) {
     // recalculate after render
     requestAnimationFrame(() => {
       this.updateVisibleRows();
+
+      const domElement = this.domElement;
+      if (!domElement) {
+        return;
+      }
+
+      // scroll to selected section if it changed
+      const {selectedSectionId} = this.props;
+      if (selectedSectionId && (selectedSectionId !== prevProps.selectedSectionId)) {
+        this.scrollToSection(selectedSectionId);
+      }
+
+      // move to current section or top of document when content switches in teacher dashboard
+      requestAnimationFrame(() => {
```

Comment:
I'm curious about the need for `requestAnimationFrame()` here. I take it a change in `props.content` doesn't otherwise trigger a re-render? Seems like there ought to be a more React-like way of accomplishing this, but it's more a matter of curiosity than a preference.

Reply:
There is something triggering scrolling after the document changes (and it isn't the mutation observer) so I needed to wait until that happened before scrolling to the appropriate section or top of document.